contracts/EntryPoint.sol

```diff
@@ -310,24 +308,27 @@ contract EntryPoint is StakeManager {
      * also make sure total validation doesn't exceed verificationGas
      * this method is called off-chain (simulateValidation()) and on-chain (from handleOps)
      */
-    function _validatePrepayment(uint256 opIndex, UserOperation calldata userOp, bytes32 requestId) private returns (uint256 requiredPreFund, PaymentMode paymentMode, bytes memory context){
+    function _validatePrepayment(uint256 opIndex, UserOperation calldata userOp, UserOpInfo[] memory opInfos) private {
```

Comment:
Just a question - can we pass an individual `UserOpInfo` here? I feel worried that this function accepts the entire array just to set a few value in a single element. Even if gas-wise this is cheap.

Reply:
Do you need to assign to 'reference'? It is already pointing somewhere, can't you just assign every field of a struct?

```
UserOpInfo memory opInfo = opInfos[opIndex];
opInfo.paymaster = paymaster;
opInfo.prepaid = prepaid;
...
```
?